### PR TITLE
aws_msk_cluster: support Cluster expansion and Open Monitoring

### PR DESCRIPTION
**Pull request by marcoreni**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes: #11215 
Closes: #10553 
Relates: #10673 (fix acceptance tests)

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_msk_cluster: support cluster expansion without cluster recreation [GH-10553]
resource/aws_msk_cluster: support changes to `enhanced_monitoring` without cluster recreation
resource/aws_msk_cluster: support `open_monitoring` configuration [GH-11215]
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSMskCluster_'
...
--- PASS: TestAccAWSMskCluster_basic (911.50s)
--- PASS: TestAccAWSMskCluster_BrokerNodeGroupInfo_EbsVolumeSize (1229.32s)
--- PASS: TestAccAWSMskCluster_EncryptionInfo_EncryptionInTransit_ClientBroker (875.43s)
--- PASS: TestAccAWSMskCluster_EncryptionInfo_EncryptionInTransit_InCluster (942.38s)
--- PASS: TestAccAWSMskCluster_EncryptionInfo_EncryptionAtRestKmsKeyArn (1005.02s)
--- PASS: TestAccAWSMskCluster_EnhancedMonitoring (1151.67s)
--- PASS: TestAccAWSMskCluster_OpenMonitoring (1300.95s)
--- PASS: TestAccAWSMskCluster_NumberOfBrokerNodes (1575.17s)
--- PASS: TestAccAWSMskCluster_Tags (876.30s)
...
```
Since enhanced monitoring is update via the same API call of open monitoring, this came free.

In order to make the acceptance tests pass I explicitly declared encryption_info configuration for all test templates. I don't know if it may be useful to write some notes inside the tests to specify this behavior.

I also fixed some tests & docs that were probably outdated (`zookeeper_connect_string` is now a list of hostnames and ports).

### CODE REVIEW

**Comment by r3kzi** [comment]
any news about this?

File: aws/resource_aws_msk_cluster.go [hunk 1]
```diff
@@ -205,7 +204,54 @@ func resourceAwsMskCluster() *schema.Resource {
 			"number_of_broker_nodes": {
 				Type:     schema.TypeInt,
 				Required: true,
-				ForceNew: true,
```
Comment (kujon) [inline aws/resource_aws_msk_cluster.go]: I believe that MSK allows you to increase, but not decrease this value. Probably due to how Kafka requires partition reassignment when the cluster make up changes.

Comment (marcoreni) [reply aws/resource_aws_msk_cluster.go]: You are right. The same applies to the broker storage size (you can expand it but you cannot decrease it). However, no special validation rules have been applied in that case, so I assume that the same can be done here.

File: aws/resource_aws_msk_cluster.go [hunk 2]
```diff
@@ -206,6 +205,54 @@ func resourceAwsMskCluster() *schema.Resource {
...
+			"open_monitoring": {
+				Type:             schema.TypeList,
+				Optional:         true,
+				DiffSuppressFunc: suppressMissingOptionalConfigurationBlock,
+				MaxItems:         1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"prometheus": {
+							Type:             schema.TypeList,
+							Optional:         true,
```
Comment (mantkiewicz) [inline aws/resource_aws_msk_cluster.go]: Is it intended to make `prometheus` optional here? In AWS schema, this is a required element https://docs.aws.amazon.com/msk/1.0/apireference/clusters.html#clusters-model-openmonitoringinfo

Comment (marcoreni) [reply aws/resource_aws_msk_cluster.go]: Fixed, thanks!

Comment (marcoreni) [reply aws/resource_aws_msk_cluster.go]: Fixed, thanks!

**Comment by anoop2811** [comment]
Hi, is there any update on this PR getting merged?

**Comment by breeves-tr** [comment]
Hello, we are also waiting on this PR to be merged.

**Comment by dennisotugo** [comment]
Hello, waiting for this. :)

File: aws/resource_aws_msk_cluster.go (hunk 1 above)
Comment (ewbankkit) [reply aws/resource_aws_msk_cluster.go]: If a `CustomizeDiff` function is added then we can `ForceNew` if the number of broker nodes is decreased.
A similar example:
https://github.com/terraform-providers/terraform-provider-aws/blob/4acecb05959a9e9b07bea9afb87d760ea887da07/aws/resource_aws_elasticsearch_domain.go#L31-L54

**Pull request merged**